backend/src/main/java/com/jobwait/persistence/PostgresController.java

```diff
@@ -153,6 +153,25 @@ public User createUserFromAuthId(String authHash) {
         }
     }
 
+    @Override
+    public List<List<Answer>> getAllAnswerSets() {
+        try {
+            Connection connection = getConnection();
+
+            String answerColumnValues = String.join(", ", Questions.knownQuestionKeys);
+            String statementText = "SELECT %s FROM answers".formatted(answerColumnValues);
+            PreparedStatement statement = connection.prepareStatement(statementText);
+
+            ResultSet resultSet = statement.executeQuery();
+            List<List<Answer>> answerSets = PersistenceUtil.resultSetRowsToAdaptedRows(resultSet,
```

Comment:
I don't think Optional fits here. The inner list represents the answers belonging to a single user, so this is a list of answer sets. I think the type makes sense here. Sure I could flatten it into a List<Answer> but maybe some stats care about the correlation between answers given by the same user (i.e. "how many full-time employees are remote?") so I think it's useful to leave it as List<List<Answer> for more information.

I agree, at first I also hated this type because nested generic classes smell funny, but I think it's grown on me. The only alternative I see here would be creating an Answers wrapper class that represents a set of Answers but I literally just removed that and I think it gets in the way of Questions being the SSOT (single source of truth) if we also have to maintain a separate Answers class too.

In conclusion, List<List<Answer>> smells funky but when the shape of the data is considered, I think it's acceptable for the time being. I hope this explanation makes you hate the type signature less.

Reply:
Ok so this is a List of answers per user, but the Answer type does not store any reference to a user so how is this distinguished at all? besides the fact that each list of answers is distinct which I guess is a somewhat useful heuristic.

I am not sure how making a wrapper class gets in the way of SSOT, SSOT does not have to mean one class stores everything, in our system answers are responses to questions, they are both within the realm of the truth, not just the questions being asked.

Reply:
> but the Answer type does not store any reference to a user so how is this distinguished at all?
Good point, but right now the concept of a User doesn't mean much besides just authentication. I don't think it will ever mean more than that. After all, any important answer we want to consider for the sake of a Stat will be implemented as another Question, not be a property on the User. So in that regard, I don't think it matters *who* answered the questions so long as we know which answers belong to the same User

> I am not sure how making a wrapper class gets in the way of SSOT
If we have an Answers class with fields for each question, we have to maintain that when we add a new question. It's no longer "update the Questions class" it's now, "update the Questions class and update the Answers class to match". Sure you might say "well Peter, it's just another line change, so what" but I say it's no longer SSOT.

Unless the Answers wrapper class doesn't have fields per question and just has a single property which is a List<Answer> which contains every answer a user made, but idk that feels redundant to me.

Reply:
> So in that regard, I don't think it matters who answered the questions so long as we know which answers belong to the same User
I agree, what I mean is that a List of List is easy to accidentally manipulate, whereas a Map of User id to List of answer would be more difficult, but this is pedantic. We are on the same page lets just leave it for now.

> Unless the Answers wrapper class doesn't have fields per question and just has a single property which is a List which contains every answer a user made, but idk that feels redundant to me.

Perhaps, I guess its not a big deal although in the end `List<Answers>` would be marginally clearer than `List<List<Answer>>`

Reply:
nvm it's a pain in the ass cause PostgresAdapter expects to return a List<A> for a type A, I can't just make it return a new wrapper class.